### ISSUE TITLE
Errors running tests after git clone and npm install from master

### ISSUE THREAD

**Issue by jharris4** [ReactTraining/react-router#6174]
The tests are failing after cloning and installing the master branch of `react-router`.

## Steps to reproduce

`git clone https://github.com/ReactTraining/react-router.git`
`npm install`
`npm run build`
`npm test`

## Expected Behavior

The tests should pass.

## Actual Behavior

The tests are failing, with the first error being:

```
FAIL modules/__tests__/NavLink-test.js
  ● NavLink › When a <NavLink> is not active › renders child components that use withRouter

    expect(received).toBe(expected) // Object.is equality
    
    Expected value to be:
      true
    Received:
      false

      213 |       );
      214 | 
    > 215 |       expect(ref instanceof WrappedComponent).toBe(true);
      216 |     });
      217 |   });
      218 | 
      
      at Object.<anonymous> (modules/__tests__/NavLink-test.js:215:47)
```

os: `MacOS 10.13.4`
node: `10.1.0`
npm: `5.6.0`

In general it seems that there are lots of issues with how dependencies are being resolved across the mono repo during test execution.

The above error is only the first in several errors that I'm seeing when trying to run the tests... :-/

**Comment by timdorr** [comment]
Are you going directly into the packages directory and running things from there? We use Lerna, which takes care of normalizing dependencies and ensures something like react-router-dom is talking to the local copy of react-router, not what's on npm. 

At the very least, run npm install and npm run build from the top level to get the dependencies correct and go from there.

**Comment by jharris4** [comment]
@timdorr perhaps I was not clear enough the above output *was from the root of the mono repo*.

My initial description omitted the `cd react-router`, but it was what I meant. The revised steps are:

```
git clone https://github.com/ReactTraining/react-router.git
cd react-router
npm install
npm run build
npm test
```

I use `lerna` with several of my own repositories, so I am well familiar with how it works and needs to be run.

Can you please confirm whether you can do the above with a clean install?

Please reopen the issue unless you can successfully do the above.

Having the tests actually work would be helpful to get my PR to `react-router-config` into a state whether the build/tests are passing... :-)

**Comment by pshrmn** [comment]
I just tried a fresh build and it looks like lerna builds the website package before the `react-router-dom` package, which causes a resolve error. The CI build has no issues with this because it skips the website. The website package has a dependency on `react-router-dom`, so lerna _should_ build `react-router-dom` first, but obviously that isn't the case.

**Comment by jharris4** [comment]
That sounds about right. Thanks for looking into this.

**Comment by pshrmn** [comment]
As I commented in #6170, the issue is that Lerna won't use the local package with an `-rc` version when the dep isn't an `-rc`. The fix should just be to make sure that all of the packages keep their deps up to date with local.

For my other lerna projects, I use `lerna publish --skip-git --skip-npm` to update versions without publishing.

**Comment by timdorr** [comment]
Hmm, I'm not having these issues locally. 

How about I just release 4.3.0? :P

**Comment by pshrmn** [comment]
I believe that you have to do a full clean (`git clean -fX`) so that `packages/react-router-dom/BrowserRouter.js` no longer exists, but a regular release would do the trick, too.

I tried running that `lerna publish` snippet I use and it didn't actually update the out of date deps, but that might be the `-rc` striking again.

As far as publshing goes, I just browser through the open PRs and I don't think any are urgent to merge.

**Comment by mjackson** [comment]
Just wanted to chime in here and say that I can confirm the tests are failing on a fresh install. Thanks for bringing this to our attention, @jharris4.

@pshrmn Are you saying we have this problem because we are currently at 4.3.0-rc.3?

**Comment by pshrmn** [comment]
@mjackson Yes, some of the packages on master have a dep of `4.2.X` and others have the `4.3.0-rc.X`; Lerna doesn't consider the ones on `4.2.2` as "local", so it doesn't produce the correct build order.

**Comment by timdorr** [comment]
I'll push out 4.3 tomorrow. I haven't heard any complaints and it'll resolve the issue.

**Comment by jharris4** [comment]
Just saw that 4.3 was released (thanks @timdorr !).

I'll see if the tests pass now, and hopefully that means I can get #6170 moving :-)
